lib/cereal/cereal.go

```diff
@@ -794,9 +793,15 @@ func (r *registeredExecutor) startTaskWorker(ctx context.Context, d backend.Task
 }
 
 func (r *registeredExecutor) runTask(ctx context.Context, t *backend.Task, taskCompleter backend.TaskCompleter) error {
-	logctx := logrus.WithField("task_name", r.name)
-
+	startTime := time.Now()
 	result, err := r.executor.Run(ctx, &task{backendTask: t})
+	endTime := time.Now()
+	logctx := logrus.WithFields(logrus.Fields{
+		"task_name":  r.name,
+		"start_time": startTime,
+		"end_time":   endTime,
+		"duration":   endTime.Sub(startTime),
+	})
 	if err != nil {
 		err := taskCompleter.Fail(err.Error())
```

Comment:
Should we also do a log `WithError` here so we can get an error message with all the context before we return?